app/authentication/resources.py

```diff
@@ -0,0 +1,40 @@
+from flask import session
+from flask_restful import Resource
+
+from app.users.models import User
+from app.lib.response_util import buildOkResponse, _returnUser
+
+
+class UserLogin(Resource):
+    """Class for logging in a user.
+    """
+    def get(self, userId):
+        user = User.query.get(userId)
+        session['id'] = user.id
+
+        return buildOkResponse(_returnUser(user))
+
+    def post(self, userId):
+        return self.get(userId)
+
+
+class UserLogOut(Resource):
+    """Class for logging out a user.
+    """
+    def get(self):
+        session.pop('id', None)
+
+        return buildOkResponse(None)
+
+
+class UserInfo(Resource):
+    """Class for accessing logged in user information.
+    """
+    def get(self):
+
+        if 'id' in session:
+            userId = session['id']
+            user = User.query.get(userId)
+            return buildOkResponse(_returnUser(user))
+        else:
+            return buildOkResponse(None)
```

Comment:
Implementing this method this way definitely works, but it would be super awesome if we didn't have to have to repeat this code for every API endpoint that we think needs authentication right? We can make something called a decorator/wrapper that we can add above any method, and its code runs before entering the `get` method. It is almost the same as calling a helper function at the top of the `get` method, but it makes it more explicit and easier to add to other methods. http://flask.pocoo.org/snippets/8/. Mostly look at `requires_auth`, and don't worry, I forget the syntax to building a decorator alll the time. We want to make our own decorator that we can add `@someDecorator` which will take care of checking the user is authenticated and raises a 401 if not. The `get` method then can just assume a user is logged in.